src/opnsense/mvc/app/views/OPNsense/CheckIP/index.volt

```diff
@@ -0,0 +1,209 @@
+are permitted provided that the following conditions are met:
+
+1.  Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2.  Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+#}
+
+<script>
+
+$( document ).ready(function() {
+
+  /**
+   * service settings
+   **/
+
+  $("#grid-services").UIBootgrid({
+    'search':'/api/checkip/settings/search/service/',
+    'get':'/api/checkip/settings/get/service/',
+    'set':'/api/checkip/settings/set/service/',
+    'add':'/api/checkip/settings/set/service/',
+    'del':'/api/checkip/settings/del/service/',
+    'toggle':'/api/checkip/settings/toggle/service/'
+  });
+
+  // Determine if at least one service is enabled as the default.  Warn if not.
+  function is_service_default() {
+    ajaxCall(url="/api/checkip/settings/is_default/service/", sendData={}, callback=function(data,status) {
+      if (status == 'success' && data['result']) {
+         $("#responseMsg").css({'background-color': '', 'color': ''});
+         $("#responseMsg").addClass('hidden');
+         $("#responseMsg").html('');
+      } else {
+         $("#responseMsg").html('<b>WARNING: </b>There is no default check IP service enabled!');
+         $("#responseMsg").css({'background-color': 'yellow', 'color': 'red'});
+         $("#responseMsg").removeClass('hidden');
+      }
+    });
+  }
+
+  // Apply changes to the DOM
+  function dom_update() {
+    // suppress FDS selection checkbox and command buttons, and keep unselected
+    $('table#grid-services tbody tr[data-row-id="FDS"]').each(function() {
+      $(this).find('td.select-cell input.select-box').addClass('hidden');
+      $(this).find('td button').filter('.command-edit,.command-copy,.command-delete').addClass('hidden');
+/**/
+      $(this).find('td.select-cell input.select-box').prop('checked', false);
+      $(this).attr('aria-selected', false);
+      $(this).removeClass('active');
+/**/
+    });
+
+    // suppress fa-times icon for verify ssl peer disabled
+    $('table#grid-services tbody tr td:nth-child(5) span').removeClass('fa-times');
```

Comment:
this selector is probably not very stable (nth-child) because the content may change and there will be some strange behaviour then.